client: panic: unaligned 64-bit atomic operation on 32-bit machine

### What version of gotd are you using?

v0.68.1

### Can this issue be reproduced with the latest version?

Yes

### What did you do?

run on 32-bit machine

### What did you expect to see?

it runs successfully.

### What did you see instead?

panic: unaligned 64-bit atomic operation.

![image](https://user-images.githubusercontent.com/61452000/190690580-373679a5-6d82-4798-8bcd-2f9658a3a8d1.png)

https://pkg.go.dev/sync/atomic#pkg-note-BUG

### What Go version and environment are you using?

v1.18

```
panic: unaligned 64-bit atomic operation

goroutine 41 [running]:
runtime/internal/atomic.panicUnaligned()
	runtime/internal/atomic/unaligned.go:8 +0x2d
runtime/internal/atomic.Xadd64(0xacb3574, 0x1)
	runtime/internal/atomic/atomic_386.s:125 +0x11
go.uber.org/atomic.(*Int64).Add(...)
	go.uber.org/atomic@v1.10.0/int64.go:50
go.uber.org/atomic.(*Int64).Inc(...)
	go.uber.org/atomic@v1.10.0/int64.go:60
github.com/gotd/td/telegram.(*Client).dc.func1()
	github.com/gotd/td@v0.68.1/telegram/pool.go:69 +0x6c
github.com/gotd/td/internal/pool.(*DC).createConnection(0xaf90780, 0x1)
	github.com/gotd/td@v0.68.1/internal/pool/pool.go:72 +0x24
github.com/gotd/td/internal/pool.(*DC).acquire(0xaf90780, {0x8ffbf4c, 0xb01e870})
	github.com/gotd/td@v0.68.1/internal/pool/pool.go:181 +0x1e8
github.com/gotd/td/internal/pool.(*DC).Invoke(0xaf90780, {0x8ffbf4c, 0xb01e870}, {0x8ff7054, 0xaf51bd8}, {0x8ff6fa0, 0xafa5100})
	github.com/gotd/td@v0.68.1/internal/pool/pool.go:246 +0x82
github.com/gotd/td/tg.(*Client).AuthImportAuthorization(0xaca9798, {0x8ffbf4c, 0xb01e870}, 0xaf51bd8)
	github.com/gotd/td@v0.68.1/tg/tl_auth_import_authorization_gen.go:204 +0x7d
github.com/gotd/td/telegram.(*Client).transfer(0xacb3340, {0x8ffbf4c, 0xb01e870}, 0xaca9798, 0x1)
	github.com/gotd/td@v0.68.1/telegram/transfer.go:30 +0xad
github.com/gotd/td/telegram.(*Client).dc(0xacb3340, {0x8ffbf4c, 0xb01e870}, 0x1, 0x1, 0xaf67ed0)
	github.com/gotd/td@v0.68.1/telegram/pool.go:97 +0x421
github.com/gotd/td/telegram.(*Client).invokeSub(0xacb3340, {0x8ffbf4c, 0xb01e870}, 0x1, {0x8ff96d8, 0xafb90a0}, {0x8ff9688, 0xafa48b8})
	github.com/gotd/td@v0.68.1/telegram/sub_conns.go:20 +0xbb
github.com/gotd/td/telegram.(*Client).invokeDirect(0xacb3340, {0x8ffbf4c, 0xb01e870}, {0x8ff96d8, 0xafb90a0}, {0x8ff9688, 0xafa48b8})
	github.com/gotd/td@v0.68.1/telegram/invoke.go:68 +0x3f5
github.com/gotd/td/telegram.InvokeFunc.Invoke(0xafa42a0, {0x8ffbf4c, 0xb01e870}, {0x8ff96d8, 0xafb90a0}, {0x8ff9688, 0xafa48b8})
	github.com/gotd/td@v0.68.1/telegram/middleware.go:15 +0x4c
github.com/gotd/contrib/middleware/floodwait.(*SimpleWaiter).Handle.func1({0x8ffbf4c, 0xb01e870}, {0x8ff96d8, 0xafb90a0}, {0x8ff9688, 0xafa48b8})
	github.com/gotd/contrib@v0.13.0/middleware/floodwait/simple_waiter.go:77 +0xa7
github.com/gotd/td/telegram.InvokeFunc.Invoke(0xaf66f40, {0x8ffbf4c, 0xb01e870}, {0x8ff96d8, 0xafb90a0}, {0x8ff9688, 0xafa48b8})
	github.com/gotd/td@v0.68.1/telegram/middleware.go:15 +0x4c
github.com/gotd/td/tg.(*Client).UploadGetFile(0xafa42a8, {0x8ffbf4c, 0xb01e870}, 0xafb90a0)
	github.com/gotd/td@v0.68.1/tg/tl_upload_get_file_gen.go:340 +0x7d
github.com/gotd/td/telegram/downloader.master.Chunk({{0x8ffc430, 0xafa42a8}, 0x1, 0x0, {0x8ffe6e4, 0xb01e780}}, {0x8ffbf4c, 0xb01e870}, 0x0, 0x80000)
	github.com/gotd/td@v0.68.1/telegram/downloader/master.go:44 +0xc7
github.com/gotd/td/telegram/downloader.(*reader).next(0xafb9000, {0x8ffbf4c, 0xb01e870}, 0x0, 0x80000)
	github.com/gotd/td@v0.68.1/telegram/downloader/reader.go:92 +0x73
github.com/gotd/td/telegram/downloader.(*reader).nextPlain(0xafb9000, {0x8ffbf4c, 0xb01e870})
	github.com/gotd/td@v0.68.1/telegram/downloader/reader.go:87 +0xc4
github.com/gotd/td/telegram/downloader.(*reader).Next(0xafb9000, {0x8ffbf4c, 0xb01e870})
	github.com/gotd/td@v0.68.1/telegram/downloader/reader.go:54 +0xe7
github.com/gotd/td/telegram/downloader.(*Downloader).parallel.func2.1({0x8ffbf4c, 0xb01e870})
	github.com/gotd/td@v0.68.1/telegram/downloader/parallel.go:50 +0xf1
github.com/gotd/td/internal/tdsync.(*CancellableGroup).Go.func1()
	github.com/gotd/td@v0.68.1/internal/tdsync/cancel_group.go:48 +0x2c
golang.org/x/sync/errgroup.(*Group).Go.func1()
	golang.org/x/sync@v0.0.0-20220819030929-7fc1605a5dde/errgroup/errgroup.go:75 +0x62
created by golang.org/x/sync/errgroup.(*Group).Go
	golang.org/x/sync@v0.0.0-20220819030929-7fc1605a5dde/errgroup/errgroup.go:72 +0xa6
```

--- a/telegram/client.go
+++ b/telegram/client.go
@@ -53,6 +53,15 @@ type clientConn interface {
 
 // Client represents a MTProto client to Telegram.
 type Client struct {
+	// Put migration in the header of the structure to ensure 64-bit alignment,
+	// otherwise it will cause the atomic operation of connsCounter to panic.
+	// DO NOT change the order of members arbitrarily.
+	// Ref: https://pkg.go.dev/sync/atomic#pkg-note-BUG
+
+	// Migration state.
+	migrationTimeout time.Duration // immutable
+	migration        chan struct{}
+
 	// tg provides RPC calls via Client. Uses invoker below.
 	tg *tg.Client // immutable
 	// invoker implements tg.Invoker on top of Client and mw.
@@ -86,9 +95,6 @@ type Client struct {
 
 	// Restart signal channel.
 	restart chan struct{} // immutable
-	// Migration state.
-	migrationTimeout time.Duration // immutable
-	migration        chan struct{}
 
 	// Connections to non-primary DC.
 	subConns    map[int]CloseInvoker

